lib/helpers/normalize-extent.js

```diff
@@ -0,0 +1,66 @@
+const joi = require('joi')
+
+const esriExtentSchema = joi.object({
+  xmin: joi.number().required(),
+  xmax: joi.number().required(),
+  ymin: joi.number().required(),
+  ymax: joi.number().required(),
+  type: joi.string().optional(),
+  spatialReference: joi.object().keys({
+    wkid: joi.number().integer().optional(),
+    latestWkid: joi.number().integer().optional()
+  }).optional()
+}).unknown()
+
+const simpleArraySchema = joi.array().items(joi.number().required()).min(4)
+const cornerArraySchema = joi.array().items(joi.array().items(joi.number()).length(2)).length(2)
+
+module.exports = function (input, spatialReference) {
+  if (!input) return undefined
+
+  const { value: arrayExtent } = validate(input, simpleArraySchema)
+
+  if (arrayExtent) {
+    return simpleArrayToEsriExtent(arrayExtent, spatialReference)
+  }
+
+  const { value: cornerArrayExtent } = validate(input, cornerArraySchema)
+
+  if (cornerArrayExtent) {
+    return cornerArrayToEsriExtent(cornerArrayExtent, spatialReference)
+  }
+
+  const { value: esriExtent } = validate(input, esriExtentSchema)
+
+  if (esriExtent) {
+    return { ...esriExtent, spatialReference }
```

Comment:
If `esriExtent.spatialReference` exists and `spatialReference` is undefined, the new extent will have undefined spatial reference. Does it matter?

Reply:
I think it does matter.  I will look.

Reply:
Same pattern exists in the pre-refactored code.  However, I would call it a bug.  Will fix.